---
title: "famsa"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{famsa}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

### Setup
First, we must load the `famsa` package into our current `R` session:

```{r setup}
library(famsa)
library(Biostrings)
```

## Usage

### The simpliest call to the `famsa` function:
Note that 'inst/examples/exampleAASequence.fasta' is an included example stringset containing Amino Acid sequences of the HBA1 protein in a group of animals sourced from https://www.ncbi.nlm.nih.gov/gene/3039/.
The returned object will be of type `AAMultipleAlignment` (in case of using DNA or RNA Stringset, the returned type will be `DNA` and `RNA` accordingly).
 
```{r, warning = TRUE, error = TRUE, message = TRUE}
exampleAAStringSet = Biostrings::readAAStringSet(system.file("extdata", "exampleAA.fasta", package="famsa"), format="fasta")
famsa(stringset=exampleAAStringSet)
```

### Setting output type to a Distance Matrix
The type of the returned object will be of a simple data.matrix representing a distance matrix.
```{r}
famsa(stringset=exampleAAStringSet, distance_matrix = TRUE)
```

## Setting output type to a phylogenetic tree
The type of the returned object will be of a phylogenetic tree in a Newick format.
Here it's plotted for better visualization
```{r, fig.show='hold', fig.width=6, fig.height=4}
plot(famsa(stringset=exampleAAStringSet, tree = TRUE))
plot(famsa(stringset=exampleAAStringSet, tree = TRUE, advanced_settings="-gt upgma"))
```
